modules/tactical/src/record.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * A Record is a version of an object stored in persistent local cache and identified by a single
+ * combination of a key and a version. The key identifies the Chain of Records for a single object
+ * and the version identifies the particular Record of that object.
+ * 
+ * Each instance of a Record requires the version that identifies it and the value of the object
+ * for that version.
+ */
+export class Record {
```

Comment:
This is a very java-esque way of writing this class. Is there a reason why we need private fields with accessors? Or can we just use a POJS object with public version and value fields?